Wikipedia/Code/WMFArticle+LegacyMobileview.swift

```diff
@@ -0,0 +1,88 @@
+
+enum MigrateMobileviewToMobileHTMLIfNecessaryError: Error {
+    case noArticleURL
+    case noArticleCacheController
+    case noMobileHTML
+}
+
+@objc extension WMFArticle {
+    
+    // TODO: use this method's completion block when loading articles (in case a mobileview conversion hasn't happened yet for that article's saved data for any reason)
+    func migrateMobileviewToMobileHTMLIfNecessary(dataStore: MWKDataStore, completionHandler: @escaping ((Error?) -> Void)) {
+        guard self.isConversionFromMobileviewNeeded == true else {
+            // If conversion was previously attempted don't try again.
+            completionHandler(nil)
+            return
+        }
+        guard let articleURL = self.url else {
+            assertionFailure("Could not get article url")
+            completionHandler(MigrateMobileviewToMobileHTMLIfNecessaryError.noArticleURL)
+            return
+        }
+
+        guard let articleCacheController = dataStore.articleCacheControllerWrapper.cacheController as? ArticleCacheController else {
+            completionHandler(MigrateMobileviewToMobileHTMLIfNecessaryError.noArticleCacheController)
+            return
+        }
+        
+        let mwkArticle = dataStore.article(with: articleURL)
+
+        dataStore.mobileviewConverter.convertMobileviewSavedDataToMobileHTML(article: mwkArticle) { (result, error) in
```

Comment:
It seems like this conversion could be entirely handled by `MWKDataStore` rather than having the model object `WMFArticle` know about the data store. Is there an advantage to this approach that I'm missing?

Reply:
You're likely right :) Will check...